wagtail/admin/wagtail_hooks.py

```diff
@@ -269,6 +270,18 @@ def register_core_features(features):
         }
     })
 
+    features.register_editor_plugin(
+        'draftail', 'br', draftail_features.BooleanFeature('enableLineBreak')
+    )
+    features.register_converter_rule('contentstate', 'br', {
+        'from_database_format': {
+            'br': LineBreakHandler(),
+        },
+        'to_database_format': {
+            'entity_decorators': {'LINE_BREAK': lambda props: DOM.create_element('br')}
+        }
```

Comment:
This `to_database_format` part probably doesn't do anything. In Draft.js line breaks are just `\n` in the text, not an entity, so [the exporter](https://github.com/springload/draftjs_exporter) processes them a specific mechanism.

You can see this here: https://github.com/wagtail/wagtail/blob/6825aab5f764cb9accfd6af47b126e8e0f12f0d2/wagtail/admin/rich_text/converters/contentstate.py#L52-L79

The `composite_decorators` part is what processes line breaks. You can see it's hard-coded, and there is no way to override it in the `for feature in self.features:` loop below. This PR needs to address that too.

Here is what the updated code might look like:

```py
        exporter_config = {
            'block_map': {
                'unstyled': 'p',
                'atomic': render_children,
                'fallback': block_fallback,
            },
            'style_map': {},
            'entity_decorators': {
                'FALLBACK': entity_fallback,
            },
            'composite_decorators': [],
            'engine': DOM.STRING,
        }

        for feature in self.features:
            rule = feature_registry.get_converter_rule('contentstate', feature)
            if rule is not None:
                feature_config = rule['to_database_format']
                exporter_config['block_map'].update(feature_config.get('block_map', {}))
                exporter_config['style_map'].update(feature_config.get('style_map', {}))
                exporter_config['entity_decorators'].update(feature_config.get('entity_decorators', {}))
                exporter_config['composite_decorators'] += feature_config.get('composite_decorators', [])
```

Then, here in `to_database_format`:

```py
        'to_database_format': {
            'composite_decorators': [
                {
                    'strategy': re.compile(r'\n'),
                    'component': br,
                },
            ],
        },
```

with the `br` definition moved here as well.